maestro-cli/src/main/java/maestro/cli/studio/DevicesService.kt

```diff
@@ -0,0 +1,74 @@
+package maestro.cli.studio
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.call
+import io.ktor.server.response.respondText
+import io.ktor.server.routing.Routing
+import io.ktor.server.routing.get
+import io.ktor.server.routing.post
+import maestro.cli.device.Device
+import maestro.cli.device.DeviceService
+import maestro.cli.session.MaestroSessionManager
+
+private data class SelectDeviceRequest(
+    val instanceId: String?,
+    val modelId: String?,
+)
+
+object DevicesService {
+    private var selectedDevice: Device.Connected? = null
+
+    fun routes(routing: Routing) {
+        routing.get("/api/devices") {
+            val connected = DeviceService.listConnectedDevices()
+            val availableForLaunch = DeviceService.listAvailableForLaunchDevices()
+            val data = Devices(connected, availableForLaunch)
+
+            val response = jacksonObjectMapper()
+                .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+                .writerWithDefaultPrettyPrinter()
+                .writeValueAsString(data)
+            call.respondText(response)
```

Comment:
Ktor can take care of serialization by itself. It does take a bit of setup: https://ktor.io/docs/serialization.html#serialization_dependency

Not a blocker, functionally your implementation would work too, just a cleaner way of doing it